internal/cmd/postgresflex/instance/clone/clone.go

```diff
@@ -0,0 +1,190 @@
+				s.Stop()
+			}
+
+			operationState := "Cloned"
+			if model.Async {
+				operationState = "Triggered cloning of"
+			}
+
+			cmd.Printf("%s instance from instance %q. New Instance ID: %s\n", operationState, instanceLabel, instanceId)
+			return nil
+		},
+	}
+	configureFlags(cmd)
+	return cmd
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().String(recoveryTimestampFlag, "", "Recovery timestamp for the instance, specified in UTC time following the format, e.g. 2024-03-12T09:28:00+00:00 ")
+	cmd.Flags().String(storageClassFlag, "", "Storage class. If not specified, storage class from the existing instance will be used.")
+	cmd.Flags().Int64(storageSizeFlag, 0, "Storage size (in GB). If not specified, storage size from the existing instance will be used.")
+
+	err := flags.MarkFlagsRequired(cmd, recoveryTimestampFlag)
+	cobra.CheckErr(err)
+}
+
+func parseInput(cmd *cobra.Command, inputArgs []string) (*inputModel, error) {
+	instanceId := inputArgs[0]
+
+	globalFlags := globalflags.Parse(cmd)
+	if globalFlags.ProjectId == "" {
+		return nil, &cliErr.ProjectIdError{}
+	}
+
+	recoveryTimestamp, err := flags.FlagToDateTimePointer(cmd, recoveryTimestampFlag, recoveryDateFormat)
+	if err != nil {
+		return nil, &cliErr.FlagValidationError{
+			Flag:    recoveryTimestampFlag,
+			Details: err.Error(),
+		}
+	}
+	recoveryTimestampString := recoveryTimestamp.Format(time.RFC3339)
+
+	return &inputModel{
+		GlobalFlagModel: globalFlags,
+		InstanceId:      instanceId,
+		StorageClass:    flags.FlagToStringPointer(cmd, storageClassFlag),
+		StorageSize:     flags.FlagToInt64Pointer(cmd, storageSizeFlag),
+		RecoveryDate:    utils.Ptr(recoveryTimestampString),
+	}, nil
+}
+
+type PostgreSQLFlexClient interface {
+	CloneInstance(ctx context.Context, projectId, instanceId string) postgresflex.ApiCloneInstanceRequest
+	GetInstanceExecute(ctx context.Context, projectId, instanceId string) (*postgresflex.InstanceResponse, error)
+	ListStoragesExecute(ctx context.Context, projectId, flavorId string) (*postgresflex.ListStoragesResponse, error)
+}
+
+func buildRequest(ctx context.Context, model *inputModel, apiClient PostgreSQLFlexClient) (postgresflex.ApiCloneInstanceRequest, error) {
+	req := apiClient.CloneInstance(ctx, model.ProjectId, model.InstanceId)
+
+	var storages *postgresflex.ListStoragesResponse
+	if model.StorageClass != nil || model.StorageSize != nil {
+		currentInstance, err := apiClient.GetInstanceExecute(ctx, model.ProjectId, model.InstanceId)
+		if err != nil {
+			return req, fmt.Errorf("get PostgreSQL Flex instance: %w", err)
+		}
+		validationFlavorId := currentInstance.Item.Flavor.Id
+
+		storages, err = apiClient.ListStoragesExecute(ctx, model.ProjectId, *validationFlavorId)
+		if err != nil {
+			return req, fmt.Errorf("get PostgreSQL Flex storages: %w", err)
+		}
+		err = postgresflexUtils.ValidateStorage(model.StorageClass, model.StorageSize, storages, *validationFlavorId)
+		if err != nil {
+			return req, err
+		}
+	}
```

Comment:
In here, it should be possible for the user to only specify the size (or only the class), right?

In the case the user only wants to change the storage class e.g., we should validate that the current instance storage size together with the new storage class are valid

Reply:
What you are doing currently is always taking `model.StorageClass` and `model.StorageSize` to `ValidateStorage`, even if one is null. Meaning this call will always fail in that case

Reply:
updated validation and adapted unit test